### DISCUSSION TITLE
Need help with snipeit upgrade , permissions failed.

### DISCUSSION THREAD

**Discussion by J1MP123** [snipe/snipe-it#14359] [Q&A]
Hello,

In past few weeks i'm playing with my snipeit installation. Problem i'm facing now is that the upgrade config says i'm lacking permissions for the folder but I don't believe I even have this sort of folder. any advice?

`--------------------------------------------------------
WELCOME TO THE SNIPE-IT UPGRADER!
--------------------------------------------------------

This script will attempt to:

- validate some very basic .env file settings
- check your PHP version and extension requirements
- check directory permissions
- do a git pull to bring you to the latest version
- run composer install to get your vendors up to date
- run migrations to get your schema up to date
- clear out old cache settings

--------------------------------------------------------
STEP 1: Checking .env file:
- Your .env is located at /var/www/snipe-it/.env
--------------------------------------------------------

√ Your APP_KEY is not blank.
√ Your APP_URL is not null or blank. It is set to http://172.16.0.110
√ Your APP_URL is set to http://172.16.0.110 and starts with the protocol (https:// or http://)
√ Your APP_URL (http://172.16.0.110) does not have a trailing slash.

--------------------------------------------------------
STEP 2: Checking PHP requirements: (Required PHP >=7.4.0 - <8.2.0)
--------------------------------------------------------

√ Current PHP version: (8.1.27) is at least 7.4.0 and less than 8.2.0! Continuing...
FYI: The php.ini used by this PHP is: /etc/php/8.1/cli/php.ini

Checking Required PHP extensions...

√ bcmath is installed!
√ curl is installed!
√ fileinfo is installed!
√ gd is installed!
√ json is installed!
√ ldap is installed!
√ mbstring is installed!
√ mysqli is installed!
√ openssl is installed!
√ PDO is installed!
√ sodium is installed!
√ tokenizer is installed!
√ xml is installed!
√ zip is installed!

--------------------------------------------------------
STEP 3: Checking directory permissions:
--------------------------------------------------------

√ /var/www/snipe-it/bootstrap/cache is writable
√ /var/www/snipe-it/storage is writable
√ /var/www/snipe-it/storage/logs is writable
√ /var/www/snipe-it/storage/logs/laravel.log is writable
√ /var/www/snipe-it/storage/framework is writable
√ /var/www/snipe-it/storage/framework/cache is writable
√ /var/www/snipe-it/storage/framework/sessions is writable
√ /var/www/snipe-it/storage/framework/views is writable
√ /var/www/snipe-it/storage/app is writable
√ /var/www/snipe-it/storage/app/backups is writable
√ /var/www/snipe-it/storage/private_uploads is writable
√ /var/www/snipe-it/public/uploads is writable

--------------------------------------------------------
The following directories/files do not seem writable:
--------------------------------------------------------
✘ PERMISSIONS ERROR: /var/www/snipe-it/storage/app/backup-temp is NOT writable
--------------------- !! ERROR !! ----------------------
Please check the permissions on the directories above and re-run this script.
------------------------- :( ---------------------------
`
![image](https://github.com/snipe/snipe-it/assets/93220034/1defd45a-a740-48e4-97d7-399bdc571123)

What am i missing, i tried multiple ways to chow...

`
   95  php upgrade.php
   96  sudo chown -R my-user:www-data /var/www/snipe-it
   97  sudo chown -R sadmin:www-data /var/www/snipe-it
   98  sudo usermod -a -G www-data sadmin
   99  sudo find /var/www/snipe-it -type f -exec chmod 664 {} \;
  100  sudo chmod -R 775 /var/www/snipe-it/storage
  101  sudo chmod -R 775 /var/www/snipe-it/public/uploads
  102  sudo chmod -R 775 /var/www/snipe-it/bootstrap/cache
  103  php upgrade.php
  104  sudo chmod -R 775 /var/www/snipe-it/public/uploads/backup-temp
  105  php upgrade.php
  106  sudo chmod -R 777 /var/www/snipe-it/public/uploads/backup-temp
  107  chown -R www-data:www-data /var/www/snipeit
  108  cd
  109  chown -R www-data:www-data /var/www/snipeit
  110  chmod -R 777 backup-temp
  111  cd /var/www/snipeit/storage/app/backup-temp
  112  cd /var/www/snipeit/storage/app/backups
  113  php upgrade.php
  114  cd var/www/snipeit
  115  cd var/www/snipe-it
  116  cd /var/www/snipe-it
  117  php upgrade.php
  118  sudo chmod -R 775 /var/www/snipe-it/storage/app
  119  php upgrade.php
  120  cd
  121  sudo passwd root
  122  reboot
  123  sudo reboot
  124  cd /var/www/snipe-it
  125  php upgrade.php
  126  cd
  127  sudo -u www-data php upgrade.php
  128  cd /var/www/snipe-it
  129  sudo -u www-data php upgrade.php
  131  php upgrade.php
sadmin@snipeit:/va`

I would really appreciate any help.

**Comment by mauro-miatello** [comment]
what's the output of
`ls -l /var/www/snipe-it/storage/app/backup-temp`
?

**Comment by J1MP123** [comment]
> what's the output of `ls -l /var/www/snipe-it/storage/app/backup-temp` ?


sadmin@snipeit:~ $ ls -l /var/www/snipe-it/storage/app/backup-temp
ls: cannot access '/var/www/snipe-it/storage/app/backup-temp': No such file or directory
sadmin@snipeit:~ $

**Comment by J1MP123** [comment]
Any idea? 🤔😔

**Comment by mauro-miatello** [comment]
have you tried to create it? if yes, same error?
